core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -100,6 +100,10 @@ public PartitionData copy() {
           found = true;
           fromProjectionPos[i] = j;
         }
+        if (fields.get(i).fieldId() == ManifestFile.SPEC_ID.fieldId()) {
+          found = true;
+          fromProjectionPos[i] = 14;
+        }
```

Comment:
This is not related to your PR but while we're here: once we find the projected value and `found` is true, do we need to iterate over the rest of the entries?

Reply:
We don't, but I don't think it's that much of a time sink